### PR TITLE
Feature/add ansible playbook

### PR DESCRIPTION
**Pull request by jhriv**
Updated the PR to use "vagrant as infrastructure" and updated the instructions to match.

"sensible-playbook" should now work as expected.

### CODE REVIEW

File: playbooks/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+*.log
```
Comment (mcritchlow) [inline playbooks/.gitignore]: Can we consolidate everything in this .gitignore into the main instance at the project root? I think I brought most things up, but may have missed a few.

**Comment by jhriv** [comment]
I merged the .gitignore, and squashed everything.

**Comment by jhriv** [comment]
... you may want to re-approve. I removed the unrelated spec/tests .gitignore entry

**Comment by mcritchlow** [comment]
Following the new install steps, `make roles` seemed to run/install all roles as expected.

Running `ansible-playbook playbook.yml` however returned
```
 [WARNING]: Host file not found: /usr/local/etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available


PLAY RECAP *********************************************************************
```

**Comment by jhriv** [comment]
Did you run "make all"? That should have created the ansible.cfg, hosts, and .ssh-config files that make "ansible-playbook" work

**Comment by lsitu** [comment]
@jhriv I ran "make all" but look looks like it's not a clean install. It's skipping installing the roles:
Downloading roles
- ucsdlib.fedora-commons is already installed, skipping.
- ucsdlib.ffmpeg is already installed, skipping.
- ucsdlib.fits is already installed, skipping.
- ucsdlib.imagemagick-jpeg2000 is already installed, skipping.
- ucsdlib.libreoffice is already installed, skipping.
- ucsdlib.postgresql is already installed, skipping.
- ucsdlib.redis is already installed, skipping.
- ucsdlib.solr is already installed, skipping.
Creating .ssh-config
...

**Comment by jhriv** [comment]
Skipping installed roles is normal. You can use "make force-roles" to re-install all of them, or "rm -rf roles/<role-to-remove>" then "make roles" to re-install the missing ones, if you want to update piece by piece (solr and ffmpeg specifically)

**Comment by mcritchlow** [comment]
Oh, I could have sworn the README said `make roles`, not `make all`. Nevermind then, I just re-ran and the VM booted up as expected. 

@lsitu - I think that's expected idempotent behavior. You already had the roles installed previously.

**Comment by lsitu** [comment]
@jhriv @mcritchlow Okay. I also see the following error like message:
==> horton: Checking for guest additions in VM...
    horton: No guest additions were detected on the base box for this VM! Guest
    horton: additions are required for forwarded ports, shared folders, host only
    horton: networking, and more. If SSH fails on this machine, please install
    horton: the guest additions and repackage the box to continue.
    horton: 
    horton: This is not an error message; everything may continue to work properly,
    horton: in which case you may ignore this message.

**Comment by jhriv** [comment]
@mcritchlow The "make roles" instruction was for vSphere/physical boxes

@lsitu that is a warning, saying that the Virtual Box extensions are missing from the VM. They are a "nice to have".  To solve this for your particular installs you can use the vagrant-vbguest plugin.
https://github.com/dotless-de/vagrant-vbguest
$ vagrant plugin install vagrant-vbguest

**Comment by lsitu** [comment]
Also, do we need another ticket to install horton app and start the development server `rake hydra:server` on the VM?

**Comment by mcritchlow** [comment]
I'm inclined to merge this since it brings across the existing roles (and them some) which was the scope of the original ticket. With future work to be done in new tickets. 

@lsitu - that work for you?

**Comment by jhriv** [comment]
I am happy with the current state - it does what it says on the tin.

**Comment by lsitu** [comment]
@mcritchlow @jhriv That's Okay for me. I think we just need to know what else we need to run the VM for demo in the next step.

**Pull request merged**